### PR TITLE
[XGAL-3] Add Idols

### PR DESCRIPTION
**Pull request by thongphp**

### CODE REVIEW

File: app/Http/Controllers/Jav/IdolController.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+use App\Repositories\JavIdolRepository;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Contracts\View\Factory;
+use Illuminate\Foundation\Auth\Access\AuthorizesRequests;
+use Illuminate\Foundation\Bus\DispatchesJobs;
+use Illuminate\Foundation\Validation\ValidatesRequests;
+use Illuminate\View\View;
+use Symfony\Component\HttpFoundation\Request;
+
+class IdolController extends BaseController
```
Comment (jooservices) [inline app/Http/Controllers/Jav/IdolController.php:23]: Please consider IdolsController instead

File: database/migrations/2020_07_18_060626_add_jav_idols_to_menu_items.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    public function up(): void
+    {
+        DB::table('menu_items')->insert(
+            [
+                [
+                    'name' => 'Jav - Idols',
+                    'link' => 'jav.idols',
+                    'type' => 'item',
+                    'icon' => null,
+                    'ordering' => 3,
```
Comment (jooservices) [inline database/migrations/2020_07_18_060626_add_jav_idols_to_menu_items.php:21]: Please use const

File: resources/views/includes/form/pagination.blade.php [hunk 3]
```diff
@@ -1,10 +1,14 @@
-<select class="custom-select form-control input-sm mr-sm-2" id="sortDir" name="sortDir">
-    <option @if(request()->request->get('sort-dir','desc') == 'asc') selected @endif value="asc">Asc</option>
-    <option @if(request()->request->get('sort-dir','desc') == 'desc') selected @endif value="desc">Desc</option>
+@php
+    $sortDirection = request()->request->get('sort-dir','desc');
+    $perPage = request()->request->get('per-page', 15);
```
Comment (jooservices) [inline resources/views/includes/form/pagination.blade.php:3]: Please move 15 to const somewhere

File: resources/views/includes/form/sort.blade.php [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <label for="sortBy"></label>
-<select class="custom-select form-control input-sm mr-sm-2" id="sortBy" name="sortBy">
+<select class="custom-select form-control input-sm mr-sm-2" id="sortBy" name="sort-by">
```
Comment (jooservices) [inline resources/views/includes/form/sort.blade.php:2]: Actually please consider correct my previous mistake by use snake name
sortBy instead sort-by.
Same with all rest

File: resources/views/jav/idols/includes/idol.blade.php [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        @include('includes.card.cover',['cover' => $item->getCover(), 'width' => '100%', 'alt' => $item->name])
+    </a>
+    <div class="card-body">
+        <a href="{{route('jav.idol.view',$item->id)}}">
+            <h5 class="card-title mr-1"><strong>{{$item->name}}</strong></h5>
+        </a>
+    </div>
+    <div class="card-footer">
+        <small class="text-muted"><i class="far fa-calendar-alt mr-1"></i>
+            @if ($item->birthday) {{$item->birthday}} @else Unknown @endif
```
Comment (jooservices) [inline resources/views/jav/idols/includes/idol.blade.php:12]: Please create method getBirthday in model with same logic. And use ->getBirthday()

File: app/Http/Controllers/Jav/IdolController.php [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    /**
+     * @param int $id
+     *
+     * @return Application|Factory|View
+     */
+    public function idol(int $id)
+    {
+        $idol = JavIdolModel::find($id);
```
Comment (jooservices) [inline app/Http/Controllers/Jav/IdolController.php:54]: What if idol now found ?

File: app/Repositories/JavIdolRepository.php [hunk 7]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Repositories;
+
+use App\Models\Jav\JavIdolModel;
+use Illuminate\Contracts\Pagination\LengthAwarePaginator;
+use Illuminate\Database\Eloquent\Builder;
+use Symfony\Component\HttpFoundation\Request;
+
+class JavIdolRepository
```
Comment (jooservices) [inline app/Repositories/JavIdolRepository.php:10]: Yay ! Remove do not extend from my previous mistake BaseRepository !
+1 for this one

File: app/Repositories/JavIdolRepository.php [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+{
+    private const DEFAULT_PAGINATION = 15;
+
+    private array $filterFields = [
```
Comment (jooservices) [inline app/Repositories/JavIdolRepository.php:14]: Actually it's more than that
`Allow search by keyword : It'll check into
LIKE %jav_idols.name%
OR LIKE %jav_idols.name%
OR LIKE %jav_idols.city%

After submitted, keyword text should be up in textbox as well ( to let user know which keyword they are using )

Allow filter age in range by use date picker

Allow filter by Measurements

Allow filter by Age

Allow change ordering by : Age / Height / Breast / Waist / Hips`

File: database/migrations/2020_07_18_060626_add_jav_idols_to_menu_items.php (hunk 2 above)
Comment (thongphp) [reply database/migrations/2020_07_18_060626_add_jav_idols_to_menu_items.php:21]: This for migration, do we really need const for row value?

Comment (jooservices) [reply database/migrations/2020_07_18_060626_add_jav_idols_to_menu_items.php:21]: Ah ah sorry. My mistake

File: app/Http/Controllers/Jav/IdolController.php (hunk 6 above)
Comment (thongphp) [reply app/Http/Controllers/Jav/IdolController.php:54]: Will redirect to `idols` page with message "idol not found"

File: app/Repositories/JavIdolRepository.php (hunk 8 above)
Comment (thongphp) [reply app/Repositories/JavIdolRepository.php:14]: This is not include in this task. We have another task for implement this. XGAL-36

**Pull request merged**